x/contracts/examples/nft/src/lib.rs

```diff
@@ -0,0 +1,688 @@
+        .expect("failed to serialize approval");
+}
+
+// /**
+//  * @dev See {IERC721-isApprovedForAll}.
+//  */
+// function isApprovedForAll(address owner, address operator) public view virtual returns (bool) {
+//     return _operatorApprovals[owner][operator];
+// }
+#[public]
+pub fn is_approved_for_all(ctx: &mut Context, owner: Address, operator: Address) -> bool {
+    ctx.get(OperatorApproval(owner, operator))
+        .expect("failed to deserialize")
+        .unwrap_or_default()
+}
+
+fn is_approved(ctx: &mut Context, operator: Address, token_id: TokenId) -> bool {
+    let owner = owner_of(ctx, token_id);
+
+    operator == owner
+        || is_approved_for_token(ctx, operator, token_id)
+        || is_approved_for_all(ctx, owner, operator)
+}
+
+fn is_approved_for_token(ctx: &mut Context, actor: Address, token_id: TokenId) -> bool {
+    ctx.get(Approval(token_id))
+        .expect("failed to deserialize")
+        .map(|approved| actor == approved)
+        .unwrap_or_default()
+}
+
+// /**
+//  * @dev See {IERC721-transferFrom}.
+//  */
+// function transferFrom(address from, address to, uint256 tokenId) public virtual {
+//     if (to == address(0)) {
+//         revert ERC721InvalidReceiver(address(0));
+//     }
+//     // Setting an "auth" arguments enables the `_isAuthorized` check which verifies that the token exists
+//     // (from != 0). Therefore, it is not needed to verify that the return value is not 0 here.
+//     address previousOwner = _update(to, tokenId, _msgSender());
+//     if (previousOwner != from) {
+//         revert ERC721IncorrectOwner(from, tokenId, previousOwner);
+//     }
+// }
+#[public]
+pub fn transfer_from(ctx: &mut Context, from: Address, to: Address, token_id: TokenId) {
+    let operator = ctx.actor();
+    let owner = owner_of(ctx, token_id);
+
+    if from != owner {
+        panic!("not the owner");
+    }
+
+    if !is_approved(ctx, operator, token_id) {
+        panic!("not approved");
+    }
+
+    let from_balance = balance_of(ctx, from);
+
+    ctx.store((
+        (Owner(token_id), to),
+        (Approval(token_id), Address::ZERO),
+        (Balance(from), from_balance - 1),
+    ))
+    .expect("failed to serialize");
+
+    if to != Address::ZERO {
+        let to_balance = balance_of(ctx, to);
+        ctx.store_by_key(Balance(to), to_balance + 1)
+            .expect("failed to serialize");
+    }
+}
+
+// /**
+//  * @dev Mints `tokenId` and transfers it to `to`.
+//  *
+//  * WARNING: Usage of this method is discouraged, use {_safeMint} whenever possible
+//  *
+//  * Requirements:
+//  *
+//  * - `tokenId` must not exist.
+//  * - `to` cannot be the zero address.
+//  *
+//  * Emits a {Transfer} event.
+//  */
+// function _mint(address to, uint256 tokenId) internal {
+//     if (to == address(0)) {
+//         revert ERC721InvalidReceiver(address(0));
+//     }
+//     address previousOwner = _update(to, tokenId, address(0));
```

Comment:
the solidity implementation overloads the functionality of the update function for no reason 🤮